test/plugins/webhooks/github.test.js

```diff
@@ -253,6 +253,16 @@ describe('github plugin test', () => {
                     assert.equal(reply.statusCode, 500);
                 });
             });
+
+            it('handles checkouting when given a non-listed user', () => {
+                userFactoryMock.get.resolves(null);
+                userFactoryMock.get.withArgs({ username: 'sd-buildbot' }).resolves(userMock);
+
+                return server.inject(options)
+                .then((response) => {
+                    assert.equal(response.statusCode, 201);
```

Comment:
You still need the `calledWith` in the other tests. The way we always have setup the stubs is like this:

``` js
   // https://github.com/screwdriver-cd/screwdriver/pull/336/files#diff-844b8ba64bd36a7e8d37f9f4d803295dR170

   userFactoryMock.get.resolves(userMock);
```

This is saying that, given any kind of input, always return `userMock`. This means that, no matter what the input is, you will always get `userMock` in return. This is why we use `assert.calledWith` in our tests: to make sure that the stub was invoked with the correct parameters because we configured the stub to not make that assertion.

Now the way that this test is setup is this:

```js
   // https://github.com/screwdriver-cd/screwdriver/pull/336/files#diff-844b8ba64bd36a7e8d37f9f4d803295dR258

  userFactoryMock.get.resolves(null);
  userFactoryMock.get.withArgs({ username: 'sd-buildbot' }).resolves(userMock);
```

This is saying that, given any kind of input, always return `null`. But, if and only if we get `{username: 'sd-buildbot}`, we return the `userMock`. Using an `assert.calledWith` is redundant here because you could never get `userMock` if the parameters were incorrect.